rclpy/src/rclpy/_rclpy.c

```diff
@@ -2971,6 +2971,191 @@ rclpy_get_node_names_and_namespaces(PyObject * Py_UNUSED(self), PyObject * args)
 }
 
 
+static rmw_ret_t
+__convert_names_and_types(
+  rcl_names_and_types_t topic_names_and_types,
+  PyObject ** pytopic_names_and_types_ptr)
+{
+  PyObject * pytopic_names_and_types = PyList_New(topic_names_and_types.names.size);
+  *pytopic_names_and_types_ptr = pytopic_names_and_types;
+  size_t i;
+  for (i = 0; i < topic_names_and_types.names.size; ++i) {
+    PyObject * pytuple = PyTuple_New(2);
+    PyTuple_SetItem(
```

Comment:
Probably want to use `PyTuple_SET_ITEM` here, and elsewhere, since this is a new tuple.